### PR TITLE
HAWKULAR-358 Add Checkstyle rule to enforce the order of Java imports

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

File: build-tools/src/main/resources/hawkular-checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -54,6 +54,13 @@
       <property name="processJavadoc" value="true" />
     </module>
 
+    <module name="ImportOrder">
+        <property name="groups" value="/^javax?\./,org,com"/>
```
Comment (pilhuhn) [inline build-tools/src/main/resources/hawkular-checkstyle/checkstyle.xml:58]: Does this ensure java, javax ordering? Or should we make that more explicit (which could be more readable)?

Comment (ppalaga) [inline build-tools/src/main/resources/hawkular-checkstyle/checkstyle.xml:58]: Yes, it does. E.g. for `org.hawkular.component.availcreator.AvailPublisher`, it enforces the order we already have in place https://github.com/hawkular/hawkular/blob/f2031fc1bfd602f708cbe77cde4a6c9102c233ef/modules/avail-creator/src/main/java/org/hawkular/component/availcreator/AvailPublisher.java#L19-L25

**Comment by ppalaga** [comment]
https://github.com/hawkular/hawkular/pull/384 shows what the introduction of this rule amounts to in Hawkular.

**Comment by ppalaga** [comment]
@pilhuhn says that https://youtrack.jetbrains.com/issue/IDEA-142468 is only about broken import of the import order prefs to Idea. The ordering itself works properly, if setup manually. Hence IDEA-142468 is not a blocker of this PR but rather a reason why we should introduce the Checkstyle rule ASAP so that contributors become aware of the need for manual settings.

**Comment by ppalaga** [comment]
projects prepared for the rule:
Hawkular: https://github.com/hawkular/hawkular/pull/384
Alerts: https://github.com/hawkular/hawkular-alerts/pull/79
Metrics: https://github.com/hawkular/hawkular-metrics/pull/296
Inventory: https://github.com/hawkular/hawkular-inventory/pull/135

**Comment by ppalaga** [comment]
Ordering the imports using Eclipse in the above projects, I have found only one subtle mismatch between Eclipse and checkstyle: Here https://github.com/hawkular/hawkular-metrics/blob/2062a0a63e3d4a9eb22f77a79e082324d3b166a1/core/metrics-core-api/src/test/java/org/hawkular/metrics/core/api/IntervalTest.java#L19-L22 

Eclipse wants

```
import static org.hawkular.metrics.core.api.Interval.parse;
import static org.hawkular.metrics.core.api.Interval.Units.DAYS;
import static org.hawkular.metrics.core.api.Interval.Units.HOURS;
import static org.hawkular.metrics.core.api.Interval.Units.MINUTES;
```

While Checkstyle wants

```
import static org.hawkular.metrics.core.api.Interval.Units.DAYS;
import static org.hawkular.metrics.core.api.Interval.Units.HOURS;
import static org.hawkular.metrics.core.api.Interval.Units.MINUTES;
import static org.hawkular.metrics.core.api.Interval.parse;
```

There was just one occurrence of this problem and I hope that we can neglect it.

**Comment by burmanm** [comment]
`@pilhuhn says that https://youtrack.jetbrains.com/issue/IDEA-142468 is only about broken import of the import order prefs to Idea`

No, the ordering is broken for import optimization no matter how you set it. Only if you manually import a single class (by pushing alt+enter) it'll work somewhat properly (but not quite).

**Comment by ppalaga** [comment]
#22 should fix the import order in IJ settings

**Pull request merged**